bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/profile/ScriptProfile.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.automation.module.script.profile;
+
+import java.util.List;
+
+import javax.script.ScriptException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.core.ConfigParser;
+import org.openhab.core.thing.profiles.ProfileCallback;
+import org.openhab.core.thing.profiles.ProfileContext;
+import org.openhab.core.thing.profiles.ProfileTypeUID;
+import org.openhab.core.thing.profiles.StateProfile;
+import org.openhab.core.transform.TransformationException;
+import org.openhab.core.transform.TransformationService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.Type;
+import org.openhab.core.types.TypeParser;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ScriptProfile} is generic profile for managing values with scripts
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class ScriptProfile implements StateProfile {
+    public static final String CONFIG_TO_ITEM_SCRIPT = "toItemScript";
+    public static final String CONFIG_TO_HANDLER_SCRIPT = "toHandlerScript";
+
+    private final Logger logger = LoggerFactory.getLogger(ScriptProfile.class);
+
+    private final ProfileCallback callback;
+    private final TransformationService transformationService;
+
+    private final List<Class<? extends State>> acceptedDataTypes;
+    private final List<Class<? extends Command>> acceptedCommandTypes;
+    private final List<Class<? extends Command>> handlerAcceptedCommandTypes;
+
+    private final String toItemScript;
+    private final String toHandlerScript;
+
+    private final boolean isConfigured;
+
+    public ScriptProfile(ProfileCallback callback, ProfileContext profileContext,
+            TransformationService transformationService) {
```

Comment:
As above, I don't really see why the profile should have a dependency on the transformation service, but maybe I am missing something crucial here.
From what I see, it seems that you are using the transformation service, as it already comes with all logic to see which script language is used and to execute it accordingly.
I'd though think it would be cleaner, if we had another config parameter, which let's the user choose the script language and then have "plain" scripts in the other config parameters (see also my "context=script" comments).
This would make the profile configuration in-line with our ScriptActions where it appears this way:

<img width="620" alt="Screenshot 2022-05-29 at 22 56 34" src="https://user-images.githubusercontent.com/3244965/170891183-1ece0830-fb06-4fe4-b5e5-bb137425bb96.png">

Wdyt?